content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md

```diff
@@ -0,0 +1,242 @@
+
+```shellsession
+#
+# Place to your jupyterhub machine (FQDN) in a group used as PBKDIR name
+#
+[production]
+127.0.0.1  ansible_connection=localhost
+```
+
+T﻿he `conf` folder hosts configuration files in a jinja format. Once expanded, the resulting files will be used by relevant workshops. I will explain in a future article all the steps and requirements to create a workshop.
+
+A﻿s part of the refacturing work to open source the project, we reaaranged the different scripts locations. We have created an install folder to handle the different installation scripts either from a Jupyterhub 's perpective or from an appliance 's standpoint too.
+
+We separated the workshops related scripts from the system ones. When one creates a workshop, one needs to provide a series of notebooks and in some cases some scripts to manage the creation, setup of a related appliance along with additional scripts to manage its lifecycle in the overall workshops-on-Demand architecture (Create, Cleanup, Reset scripts at deployment or Cleanup times). These scripts need to be located in the `scripts` folder. On the other hand, the system scripts are located in the `sys` folder.
+
+![](/img/tree-wkshop2.png "Tree view of the sys directory")
+
+T﻿his directory hosts important configuration files for both the system and Jupyterhub. You can see for instance `fail2ban` configuration files. Some jinja templates are present here too. These templates will be expanded trough the `deliver` mechanism allowing the creation of files customized with ansible variables. All the wod related tasks are prefixed with wod for better understanding and ease of use.
+
+They can refer to some Jupyterhub kernel needs like `wod-build-evcxr.sh.j2` that aims at creating a script allowing the rust kernel installation. Some other templates are related to the system and Jupyterhub. `wod-kill-processes.pl.j2` has been created after discovering the harsh reality of online mining...In a ideal world, I would not have to explain further as the script would not be needed. Unfortunately, this is not the case, when one offers access to some hardware freely online, he can expect sooner or later to  see his original and pure idea to be highjacked...Let's say that you want to provide some AI/ML 101 type of workshops, you may consider providing servers with some GPUs. Any twisted minded cryptominer discovering your ressources will definitely think he hits the jackpot! This little anecdot actually happened to us and not only on GPU based servers, some regular servers got hit as well. We found out that performance on some servers became very poor and when looking into it, we found some scripts that were not supposed to run there...As a result, we implemented monitors to check load on our server and make sure that to  kill any suspicious process before kicking out the misbehaving student.
```

Comment:
What do you mean by "They"? The Jina templates? The wod-related tasks? It's best to spell it out at the beginning of a paragraph so the reader is clear on what the subject is here.

JupyterHub kernel.

Replace the elipsis (three dots) with a period.

Modify the section that reads "Unfortunately..." to:
"Unfortunately, this is not the case. When one offers access to hardware freely online, sooner or later, he can expect to see his original idea to be hyjacked.  

This is a very long paragraph and it would be easier to read if it were broken out into two paragraphs, but I'm not quite sure where the best break is. I'll leave you to decide that.

Don't use the elipsis. You have it at least 3 times in this paragraph. Just go with simple sentences, ending them each with a period.

It might be good to start the second paragraph right after "highjacked". Then, start the next line with a modified version of what you have:
"Let's say that you want to provide some AI/ML 101-type of workshops. As part of this, you may consider providing servers with GPUs. Any twisted-minded cryptominer discovering your resources will definitely think that he's hit the jackpot!"  (note that "resources" had a typo)

The last sentence should be modified to read:
"As a result, we implemented monitors to check the load on our server and made sure we killed any suspicious processes before kicking out the misbehaving student."  [I love this, by the way... great story!]